apps/homework/tasks.py

```diff
@@ -38,21 +39,41 @@ def retrieve_score_from_jupyter_notebook(fd, submission):
     # If there are multiple matches, a warning is placed on the submission, and the last match in the file is used
     # per the specs of this feature.
 
+    existing_messages = submission.reporting_messages
+    if fd.name.split('.')[-1] != 'ipynb':
+        if 'errors' not in existing_messages:
+            existing_messages['errors'] = []
+        existing_messages['errors'].append('File submitted does not have .ipynb extension. (Hint that this is not a Jupyter Notebook.')
+        logger.error(f'The file submitted with this notebook does not have .ipynb extension. Error in submission {submission.pk} for homework "{submission.definition.name}" (pk={submission.definition.pk}).')
+
     filtered_content = list(filter(lambda line: 'Your final score is' in str(line) and 'print' not in str(line), content))
-    if len(filtered_content) is not 1:
+    if len(filtered_content) != 1:
         # Add warning to the submission model
-        logger.warning(f'Multiple Jupyter Notebook score string matches in submission {submission.pk} for homework "{submission.definition.name}" (pk={submission.definition.pk}). Exactly one line should contain the phrase "final score" and not include the phrase "print".')
+        if len(filtered_content) > 1:
+            if 'warnings' not in existing_messages:
+                existing_messages['warnings'] = []
+            existing_messages['warnings'].append('Multiple Jupyter Notebook score string matches in submission. Only one line should contain the matching score string, "Your final score is x / x, congratulations!".')
+            logger.warning(f'Multiple Jupyter Notebook score string matches in submission {submission.pk} for homework "{submission.definition.name}" (pk={submission.definition.pk}). Exactly one line should contain the phrase "final score" and not include the phrase "print".')
+        if len(filtered_content) == 0:
+            if 'errors' not in existing_messages:
+                existing_messages['errors'] = []
+            existing_messages['errors'].append('Cannot find score in Jupyter Notebook. No matches for the string, "Your final score is x / x, congratulations!".')
+            logger.error(f'Cannot find score in Jupyter Notebook. No matches for the string, "Your final score is x / x, congratulations!". Error in submission {submission.pk} for homework "{submission.definition.name}" (pk={submission.definition.pk}).')
+
+    submission.reporting_messages = existing_messages
+    submission.save()
 
-    score_string = str(filtered_content[-1])
-    start_index = score_string.find('is')
+    if len(filtered_content) > 0:
+        score_string = str(filtered_content[-1])
+        start_index = score_string.find('is')
```

Comment:
what is this doing?

Reply:
I think this is a remnant of an old solution. Thanks for catching it! ✅